text/0060-refashion-updates-apis.md

```diff
@@ -0,0 +1,416 @@
+
+- The `update` resource is renamed `task`. The names of existing API routes are also changed to reflect this change.
+- Tasks are now also accessible as an independent resource of an index. `GET - /tasks`; `GET - /tasks/:taskUid`
+- A `task_not_found` error is introduced.
+-  The format of the `task` object is updated.
+    - `updateId` becomes `uid`.
+    - Attributes of an error appearing in a `failed` `task` are now contained in a dedicated `error` object.
+    - `type` is no longer an object. It now becomes a string containing the values of its `name` field previously defined in the `type` object.
+    - The possible values for the `type` field are reworked to be more clear and consistent with our naming rules.
+    - A `details` object is added to contain specific information related to a `task` payload that was previously displayed in the `type` nested object.
+    - An `indexUid` field is added to give information about the related index on which the task is performed.
+    - `duration` format has been updated to express an `ISO 8601` duration.
+    - `processed` status changes to `succeeded`.
+    - `startedProcessingAt` is updated to `startedAt`.
+    - `processedAt` is updated to `finishedAt`.
+- `202 Accepted` requests previously returning an `updateId` are now returning a summarized `task` object.
+- `MEILI_MAX_UDB_SIZE` env var is updated `MEILI_MAX_TASK_DB_SIZE`.
+- `--max-udb-size` cli option is updated to `--max-task-db-size`.
+
+
+### II. Motivation
+
+The main motivation is to stabilize the current `update` resource to a version that conforms to our API convention and thus allow future evolutions on a more solid base. We would like to modify the name `update`, its format is also to be reviewed because some attributes are not immediately clear, either in the possible values or in the chosen names.
+
+### III. Explanation
+
+#### 1. `task` object definition
+
+##### **Fully Qualified `task` object**
+
+> This fully qualified version appears as response object on `task` dedicated endpoints.
+
+| field   | type    | description                     |
+|---------|---------|---------------------------------|
+| uid      | integer | Unique sequential identifier           |
+| indexUid | string | Unique index identifier |
+| status  | string  | Status of the task. Possible values are `enqueued`, `processing`, `succeeded`, `failed`                                |
+| type    | string  | Type of the task. Possible values are `documentsAddition`, `documentsPartial`, `documentsDeletion`, `settingsUpdate`, `clearAll` |
+| details | object |  Details information of the task payload. See `details` object definition by `type` part. |
+| error | object | Error object containing error details and context when a task has a `failed` status. See https://github.com/meilisearch/specifications/pull/61|
+| duration | string | Total elasped time the engine was in processing state expressed as a `ISO-8601` duration format. Default is set to `null`.  |
```

Comment:
sorry, couldn't help it.
```suggestion
 | duration | string | Total elapsed time the engine was in processing state expressed as a `ISO-8601` duration format. Default is set to `null`.  |
```

Reply:
Thanks @guimachiavelli 😇

By the way, what do you think of this nomenclature for the `duration` field? e.g. `PT3M6S` to express a duration of 3 minutes and 6 seconds.

Reply:
Hmmm…do you mean using `PT[number]M[number]S` instead of `hh:mm:ss` (which is what I understood ISO-8601 stood for)? I personally find it a bit hard to read. Like, I need a reference to understand what it stands for since it's not a standard (or not a standard I have previously encountered.) Or perhaps I could figure it out on my own, but it takes a bit more brain processing time.

To give my full (very personal) preference, I usually prefer duration in APIs to be given in seconds, as it makes it easier to calculate things. I'm a fan of UNIX epoch for the same reason. This is probably because of my background in front-end, so it is often the case you need to do stuff like e.g. compensating for the duration of a task, updating animations, etc based on time elapsed.